client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java

```diff
@@ -306,11 +307,15 @@ private HttpRequest createHttpRequest(SwaggerMethodParser methodParser, Object[]
                 final String bodyContentString = serializer.serialize(bodyContentObject, bodyEncoding(request.headers()));
                 request.withBody(bodyContentString, contentType);
             }
+            else if (bodyContentObject instanceof AsyncInputStream) {
+                AsyncInputStream stream = (AsyncInputStream) bodyContentObject;
+                request.withBody(new FlowableHttpRequestBody(stream.contentLength(), contentType, stream.content()));
+            }
             else if (bodyContentObject instanceof byte[]) {
                 request.withBody((byte[]) bodyContentObject, contentType);
             }
             else if (bodyContentObject instanceof FileSegment) {
-                request.withBody(new FileRequestBody((FileSegment) bodyContentObject));
+                throw new UnsupportedOperationException("FileSegment/zero-copy I/O is not currently supported.");
             }
```

Comment:
If we're not going to support it right now, then I'd suggest completely removing the FileSegment class. We can always look in the history if we need to look up how the implementation worked. It might be worth adding a tag for this commit in git that specifies that the implementation of FileSegment can be found here.

Reply:
That sounds like the right thing to do.